PyriteEditor/src/editorPasses/PickerPass.h

```diff
@@ -0,0 +1,502 @@
+                    }
+
+                    pBillboardIDBuffer->setData(
+                        BillboardPickerIDBuffer::data_t{ .ids = ids[0] });
+                    
+
+                	m_pickEffect_Billboards->bindConstantBuffer("CameraBuffer", pcameraBuffer);
+                    m_pickEffect_Billboards->bindConstantBuffer("BillboardPickerIDBuffer", pBillboardIDBuffer);
+
+                    std::vector<pyr::Texture> sortedTextures;
+                    sortedTextures.resize(16);
+                    for (const auto& [texPtr, texId] : renderData.textures)
+                    {
+                        sortedTextures[texId] = *texPtr;
+                    }
+                    m_pickEffect_Billboards->bindTextures(sortedTextures, "textures");
+
+                    m_pickEffect_Billboards->bind();
+                    renderData.instanceBuffer.bind(true);
+                    pyr::Engine::d3dcontext().DrawInstanced(6, static_cast<UINT>(renderData.instanceBuffer.getVerticesCount()), 0, 0);
+                    m_pickEffect_Billboards->unbindResources();
+                }
+
+                m_idTarget.unbind();
+
+                pyr::RenderProfiles::popDepthProfile();
+
+                int actorId = ReadActorIDFromTexture();
+                if (actorId != ID_NONE)
+                {
+                    // You have picked actor !!!
+                    PYR_LOGF(LogRenderPass, INFO, "Actor {} was picked.", actorId);
+                    AddSelectedActor(actorId, bIsHoldingControl);
+
+                }
+                if (renderDoc)
+                    renderDoc->EndFrameCapture(nullptr, nullptr);
+            }
+
+            /// Creates a staging texture to readback from the 1x1 target.
+            /// Returns the editor actor ID, or ID_NONE (== -1) if the operation fails.
+            /// Note that editor ID start at 1, so reading 0 is valid and corresponds to basically no actor.
+            int ReadActorIDFromTexture() {
+            
+                pyr::Texture sourceTexture = m_idTarget.getTargetAsTexture(pyr::FrameBuffer::COLOR_0);
+
+                D3D11_SHADER_RESOURCE_VIEW_DESC sourceDesc{};
+                sourceTexture.getRawTexture()->GetDesc(&sourceDesc);
+
+                // -- Create a staging texture with the format of the source texture
+                ID3D11Texture2D* StagingTexture;
+                D3D11_TEXTURE2D_DESC srcDesc;
+                ZeroMemory(&srcDesc, sizeof(srcDesc));
+                srcDesc.Width = 1;
+                srcDesc.Height = 1;
+                srcDesc.ArraySize = 1;
+                srcDesc.MipLevels = 0;
+                srcDesc.Format = DXGI_FORMAT_R32G32B32A32_FLOAT;
+                srcDesc.SampleDesc.Count = 1;
+                srcDesc.SampleDesc.Quality = 0;
+                srcDesc.Usage = D3D11_USAGE_STAGING;
+                srcDesc.CPUAccessFlags = D3D11_CPU_ACCESS_READ;
+                srcDesc.BindFlags = 0;
+                srcDesc.MiscFlags = 0;
```

Comment:
You probably don't want to create the texture/render target each time, making it on construction should be simpler